rls/src/main/java/io/grpc/rls/internal/LbPolicyConfiguration.java

```diff
@@ -0,0 +1,372 @@
+          new ChildLoadBalancingPolicy(
+              childPolicyConfigTargetFieldName,
+              (Map<String, Object>) effectiveChildPolicy.values().iterator().next(),
+              effectiveLbProvider);
+    }
+
+    /** Creates a child load balancer config for given target from elected raw child policy. */
+    public Map<String, ?> getEffectiveChildPolicy(String target) {
+      Map<String, Object> childPolicy = new HashMap<>(effectiveRawChildPolicy);
+      childPolicy.put(childPolicyConfigTargetFieldName, target);
+      return childPolicy;
+    }
+
+    /** Returns the elected child {@link LoadBalancerProvider}. */
+    public LoadBalancerProvider getEffectiveLbProvider() {
+      return effectiveLbProvider;
+    }
+
+    void addPendingRequest(RouteLookupRequest request, BackoffPolicy backoffPolicy) {
+      checkNotNull(request, "request");
+      checkNotNull(backoffPolicy, "backoffPolicy");
+      BackoffPolicy existing = pendingRequests.put(request, backoffPolicy);
+      checkState(
+          existing == null,
+          "This is a bug, there should be at most one outstanding pending request");
+    }
+
+    void removePendingRequest(RouteLookupRequest request) {
+      BackoffPolicy policy = pendingRequests.remove(request);
+      checkState(policy != null, "This is a bug, untracked pending request found");
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      ChildLoadBalancingPolicy that = (ChildLoadBalancingPolicy) o;
+      return Objects.equals(effectiveRawChildPolicy, that.effectiveRawChildPolicy)
+          && Objects.equals(effectiveLbProvider, that.effectiveLbProvider)
+          && Objects.equals(childPolicyConfigTargetFieldName, that.childPolicyConfigTargetFieldName)
+          && Objects.equals(pendingRequests, that.pendingRequests);
+    }
+
+    @Override
+    public int hashCode() {
+      return
+          Objects
+              .hash(
+                  effectiveRawChildPolicy,
+                  effectiveLbProvider,
+                  childPolicyConfigTargetFieldName,
+                  pendingRequests);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("effectiveRawChildPolicy", effectiveRawChildPolicy)
+          .add("effectiveLbProvider", effectiveLbProvider)
+          .add("childPolicyConfigTargetFieldName", childPolicyConfigTargetFieldName)
+          .add("pendingRequests", pendingRequests)
+          .toString();
+    }
+  }
+
+  /**
+   * ChildPolicyWrapper is a wrapper class for child load balancing policy with associated helper /
+   * utility classes to manage the child policy.
+   */
+  static final class ChildPolicyWrapper implements Closeable {
+
+    @VisibleForTesting
+    static final Map<String /* target */, RefCountedChildPolicyWrapper> childPolicyMap =
```

Comment:
This map is shared for the whole JVM for all channels to totally different servers. This worries me.  Also the map needs synchronization.